### DISCUSSION TITLE
How can i remove equals sign "=" when using rewrites

### DISCUSSION THREAD

**Discussion by Jiny3213** [vercel/next.js#73576] [Help]
### Summary

I was using a old third-party oss image services, it can set quality by adding `?imageView2/2/w/200/q/55`. But `?imageView2/2/w/200/q/55=` will cause an error. I want to use rewrites as a proxy to the oss server.

Here is my `next.config.js`
```
rewrites: async () => {
    return [
      {
        source: '/static/:path',
        destination: 'http:/my-oss-domain.com/:path',
      },
    ]
  }
```

I want to rewrite `/static/asdasdasd?imageView2/2/w/200/q/55` to `http:/my-oss-domain.com/static/asdasdasd?imageView2/2/w/200/q/55`, but it always go to `http:/my-oss-domain.com/static/asdasdasd?imageView2/2/w/200/q/55=` and cause an error.

I also try `middleware.ts`
```
function middleware(req) {
  if(req.url.includes('/static')) {
    const { pathname, search } = new URL(req.nextUrl)
    return NextResponse.rewrite(`http:/my-oss-domain.com${pathname.replace('/static', '')}${search.replace('=', '')}`)
  }
}
```
But it also add a "=" after that, cause an error.

**Comment by icyJoseph** [comment]
I think the issue is that, the moment you have a `?` then a query (url search params) starts. Could you point to the docs where they document, `?imageView2/2/w/200/q/55`, on your image service?

in other words, you have `key=value`, where `key` is `?imageView2/2/w/200/q/55`, and value is empty.

**Reply by Jiny3213** [reply]
Here is the document in Chinese, they has no English version https://developer.qiniu.com/dora/1279/basic-processing-images-imageview2

It's strange to has only `key`, may be a history problem. (Their service started in 2012)

**Reply by icyJoseph** [reply]
Mmm, I see, the issue is the `toString` method on `URLSearchParams` (web API):

```js
let url = 'https://dora-doc.qiniu.com/gogopher.jpg?imageView2/1/w/200/h/200';
new URLSearchParams(new URL(url).search).toString(); // 'imageView2%2F1%2Fw%2F200%2Fh%2F200=
new URL(url).searchParams.toString() // 'imageView2%2F1%2Fw%2F200%2Fh%2F200='
```

Either approach on that web API yields a `=` at the end... 🤔 ~ is this per spec?

Safari, Chrome, FireFox, Deno and Node, also yield the `=` at the end.

```console
Deno 2.0.6
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> new URLSearchParams(new URL('https://dora-doc.qiniu.com/gogopher.jpg?imageView2/1/w/200/h/200').search).toString()
"imageView2%2F1%2Fw%2F200%2Fh%2F200="
```

and

```console
➜ node
Welcome to Node.js v18.18.0.
Type ".help" for more information.
> new URLSearchParams(new URL('https://dora-doc.qiniu.com/gogopher.jpg?imageView2/1/w/200/h/200').search).toString()
'imageView2%2F1%2Fw%2F200%2Fh%2F200='
> .exit
```

**Reply by icyJoseph** [reply]
My guess is that API is somehow used and causes the `=` to be appended, but I have no way forward for you

**Reply by Jiny3213** [reply]
Thank you very mach!!! 😊
